hive/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -188,6 +338,41 @@ protected String defaultWarehouseLocation(TableIdentifier tableIdentifier) {
         tableIdentifier.name());
   }
 
+  public  Map<String, String> convertToMetadata(Database database) {
+    return ImmutableMap.<String, String>builder()
+        .put("location", database.getLocationUri())
+        .put("comment", database.getDescription())
+        .putAll(database.getParameters())
+        .build();
+  }
+
+  public  Database convertToDatabase(Namespace namespace, Map<String, String> meta) {
+    String warehouseLocation = conf.get("hive.metastore.warehouse.dir");
+
+    if (namespace.levels().length != 1) {
+      throw new NoSuchNamespaceException("Namespace does not exist: %s", namespace);
+    }
+
+    Database database  = new Database();
+    Map<String, String> parameter = Maps.newHashMap();
+
+    database.setName(namespace.toString());
+    database.setLocationUri(SLASH.join(warehouseLocation, namespace.level(0)) + ".db");
```

Comment:
Can this use `new Path(warehouseLocation, namespace.level(0)).toString()` instead of using `SLASH.join`?